From 5e715100bdc763e84232bff6cc82bf4821a13ae1 Mon Sep 17 00:00:00 2001
From: Brendan King <Brendan.King@imgtec.com>
Date: Tue, 25 Aug 2020 14:12:32 +0100
Subject: [PATCH 36/59] egl/wayland: process non-resized window movement

The dx and dy parameters to the wl_egl_window_resize function were
not being processed unless the window width or height were being
changed.
---
 src/egl/drivers/dri2/platform_wayland.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/egl/drivers/dri2/platform_wayland.c b/src/egl/drivers/dri2/platform_wayland.c
index f10531e1c72..e2643efde20 100644
--- a/src/egl/drivers/dri2/platform_wayland.c
+++ b/src/egl/drivers/dri2/platform_wayland.c
@@ -391,6 +391,9 @@ resize_callback(struct wl_egl_window *wl_win, void *data)
    struct dri2_egl_display *dri2_dpy =
       dri2_egl_display(dri2_surf->base.Resource.Display);
 
+   dri2_surf->dx = wl_win->dx;
+   dri2_surf->dy = wl_win->dy;
+
    if (dri2_surf->base.Width == wl_win->width &&
        dri2_surf->base.Height == wl_win->height)
       return;
@@ -1243,8 +1246,6 @@ update_buffers(struct dri2_egl_display *dri2_dpy,
 
       dri2_surf->base.Width  = dri2_surf->wl_win->width;
       dri2_surf->base.Height = dri2_surf->wl_win->height;
-      dri2_surf->dx = dri2_surf->wl_win->dx;
-      dri2_surf->dy = dri2_surf->wl_win->dy;
    }
 
    if (dri2_surf->resized || dri2_surf->received_dmabuf_feedback) {
-- 
2.25.1

